src/node/desktop/src/main/session-launcher.ts

```diff
@@ -191,4 +193,34 @@ export class SessionLauncher {
   closeAllSatellites(): void {
     console.log('CloseAllSatellites not implemented');
   }
+
+  getLauncherToken(): string {
+    if (!this.launcherToken) {
+      this.launcherToken = generateShortenedUuid();
+    }
+    return this.launcherToken;
+  }
+
+  getPort(): string {
```

Comment:
It feels a little odd that a function that looks like a plain 'getter' also sets `this.port` as a side-effect (and the associated environment variable).

Reply:
Agree, just doing a literal port here.

Reply:
Gotcha. IMHO for cases like these, we can re-examine some of the old decisions and try to clean up the implementation where appropriate; we don't necessarily need a direct port. Does that make sense, or would you prefer tracking the old implementation as closely as possible?

Reply:
Well, not a strictly literal port, have moved this out of the Options() class and put ownership into the session-launcher. I'll do some tidying up on this before I merge. Agree too many side-effects here!

Reply:
@kevinushey No, I agree we should cleanup as we go along, within reason.